cli/src/jvmMain/kotlin/komem/litmus/CliJvm.kt

```diff
@@ -0,0 +1,64 @@
+package komem.litmus
+
+import com.github.ajalt.clikt.parameters.options.*
+import com.github.ajalt.clikt.parameters.types.choice
+import jcstressDirectory
+import komem.litmus.barriers.JvmSpinBarrier
+
+class CliJvm : CliCommon() {
+    override val runner by option("-r", "--runner")
+        .choice(mapOf("thread" to JvmThreadRunner, "jcstress" to JCStressRunner))
+        .required()
+    override val barrierProducer = ::JvmSpinBarrier
+    override val affinityMapSchedule = listOf(null)
+
+    private val allowJCStressReruns by option("--allow-jcs-reruns")
+        .flag()
+    private val jcstressFreeArgs by option("-j", "--jcsargs")
+        .convert { it.split(" ") }
+        .default(emptyList())
+
+    override fun run() = if (runner == JCStressRunner) jcstressRun() else super.run()
+
+    private fun jcstressRun() {
+        println("haha: $jcstressFreeArgs")
+
+        val paramsList = variateRunParams(
+            batchSizeSchedule = batchSizeSchedule,
+            affinityMapSchedule = affinityMapSchedule,
+            syncPeriodSchedule = syncEverySchedule,
+            barrierSchedule = listOf(barrierProducer),
+        ).toList()
+        when (paramsList.size) {
+            0 -> {
+                echo("parameters list is empty; ensure no empty lists are used", err = true)
+                return
+            }
+
+            1 -> {} // ok
+            else -> {
+                if (!allowJCStressReruns) {
+                    echo(
+                        "you likely don't want to run JCStress multiple times;" +
+                                " if you're sure, enable --allow-jcs-reruns",
```

Comment:
why you don't want that?

Reply:
JCStress can variate its parameters on its own, and it probably does it very well in some tricky way, so it seems very unlikely to me that someone would really want to run it multiple times with different manually-set parameters